src/conn_sock.c

```diff
@@ -161,21 +161,35 @@ static char *bind_unix_socket(char *socket_relative_name, int sock_type, mode_t
 		get_sock_path(use_full_attach_path, g_build_filename(opt_socket_path, opt_cuuid, NULL), sizeof(socket_addr.sun_path));
 
 	/*
-	 * Instead of creating a symlink, open the desired socket path and use the fd reference
-	 * from /proc/self/fd.
+	 * Create a symlink to match expectations of callers, even though we no longer
+	 * worry about exceeding the unix socket path limit.
```

Comment:
could we create the symlink only when `!use_full_attach_path`?

Reply:
we need this symlink regardless, as podman references `socket_path/attach` but the attach file actually lives in `bundle_path/attach`

Reply:
unless you're referring to dropping the sock_path entirely with this PR, and then do so in podman when we use --full-attach

Reply:
done, now callers who use `--full-attach` should not rely on  the socket dir being used (looking at you, podman)